persistence-cassandra-3.11/src/main/java/io/stargate/db/cassandra/impl/CassandraPersistence.java

```diff
@@ -110,6 +113,15 @@ public void initialize(Config config)
         daemon = new CassandraDaemon(true);
 
         DatabaseDescriptor.daemonInitialization(() -> config);
+
+        root = new InternalDataStore();
+        authenticator = new AuthenticatorWrapper(DatabaseDescriptor.getAuthenticator());
+        handler = org.apache.cassandra.service.ClientState.getCQLQueryHandler();
+        interceptor = new DefaultQueryInterceptor();
+        interceptor.initialize();
```

Comment:
Move this initialization before because if a `IEndpointStateChangeSubscriber` isn't registered before `Gossiper.start()` the events might be missed.